first-read-last-write-cache/src/cache.rs

```diff
@@ -130,24 +130,37 @@ impl CacheLog {
     ///     k1 => v1.merge(v1') <- preserves the first read and the last write for 'k1'
     ///     k2 => v2
     ///     k3 => v3
-    pub fn merge(self, mut rhs: Self) -> Result<Self, MergeError> {
-        let mut new_cache = CacheLog::with_capacity(self.log.len() + rhs.log.len());
-
-        for (self_key, self_access) in self.log {
-            match rhs.log.remove(&self_key) {
-                Some(rhs_access) => {
+    pub fn merge(&mut self, rhs: &mut Self) -> Result<(), MergeError> {
```

Comment:
What about `merge_left`?

Reply:
It's not necessarily clear to me what `merge_left` means on first glance (and I'd expect there to be a corresponding `merge_right` function), but at least it will make people stop and read the docs!

We can go with it for now, and keep thinking about good names

Reply:
Changed to `merge_left`